### DISCUSSION TITLE
Non-partitioned asset downstream from a partitioned asset failure

### DISCUSSION THREAD

**Discussion by steffyd** [dagster-io/dagster#18818] [Q&A] [answered]
I have a partitioned asset, and a downstream asset that has a LastPartitionMapping as the AssetIn. The partitioned asset has an io manager that accesses the context.upstream_output.partition_key in load_input, but when I try to materialize the downstream asset I get an error
dagster._check.CheckError: Failure condition: Tried to access partition_key on a non-partitioned run.
How do I properly connect these two assets?

#ask-ai responded that I needed to materialize the downstream asset with a specific partition key, but the downstream asset doesn't have partitions? Am I missing something that I need to do regarding this?

**Comment by steffyd** [comment, marked as answer]
I realized I was using the wrong partition determination in load_input by trying to access the upstream_output parititon_key

**Comment by Joe-Koch** [comment]
I got the same error using the DuckDBPandasIOManager and a multi-partitioned asset upstream of a non-partitioned asset. A workaround was to access the upstream asset with a SQL query instead of a pandas dataframe directly. E.g. change 
```
@asset
def downstream_asset(
    upstream_asset: pd.DataFrame
) -> pd.DataFrame:
    df = upstream_asset
```
to 
```
@asset(deps=[upstream_asset])
def downstream_asset(
    database: Database,
) -> pd.DataFrame:
    df = database.query("SELECT * FROM upstream_asset")
```

This was weird because [the iris example from the docs](https://docs.dagster.io/integrations/duckdb/reference#storing-partitioned-assets) still worked for me. 
```
@asset(
    partitions_def=StaticPartitionsDefinition(
        ["Iris-setosa", "Iris-virginica", "Iris-versicolor"]
    ),
    metadata={"partition_expr": "SPECIES"},
)
def iris_dataset_partitioned(context: AssetExecutionContext) -> pd.DataFrame:
    species = context.partition_key

    full_df = pd.read_csv(
        "https://docs.dagster.io/assets/iris.csv",
        names=[
            "sepal_length_cm",
            "sepal_width_cm",
            "petal_length_cm",
            "petal_width_cm",
            "species",
        ],
    )

    return full_df[full_df["species"] == species]


@asset
def iris_cleaned(iris_dataset_partitioned: pd.DataFrame):
    return iris_dataset_partitioned.dropna().drop_duplicates()
```
